manifests/agent/install.pp

```diff
@@ -1,8 +1,7 @@
 # Install the puppet client installation
 class puppet::agent::install {
-
-  package { $puppet::params::client_package:
-    ensure => $::puppet::version,
+  package { $puppet::client_package:
+    ensure   => $::puppet::version,
+    provider => $::puppet::package_provider,
```

Comment:
@ekohl - I learned something new today thanks to this... didn't know it would work with `provider => undef`